src/sqlancer/clickhouse/gen/ClickHouseCommon.java

```diff
@@ -0,0 +1,156 @@
+    }
+
+    public static String createIndexName(int nr) {
+        return String.format("i%d", nr);
+    }
+
+    public static String getCheckConstraint(ClickHouseProvider.ClickHouseGlobalState globalState, List<ClickHouseSchema.ClickHouseColumn> columns) {
+        ClickHouseExpression expression = new ClickHouseExpressionGenerator(globalState).setColumns(columns)
+                .generateExpression(new ClickHouseSchema.ClickHouseLancerDataType(ClickHouseDataType.Int32));
+        return (" CHECK ( " + ClickHouseVisitor.asString(expression) + ")");
+    }
+
+    public static ClickHouseExpression getTrueExpression(List<ClickHouseSchema.ClickHouseColumn> columns, ClickHouseProvider.ClickHouseGlobalState globalState) {
+        ClickHouseExpression randomExpression = new ClickHouseExpressionGenerator(globalState).setColumns(columns)
+                .generateExpression(new ClickHouseSchema.ClickHouseLancerDataType(ClickHouseDataType.Int32));
+        ClickHouseConstant expectedValue = randomExpression.getExpectedValue();
+        if (expectedValue == null) {
+            throw new IgnoreMeException();
+        }
+        Optional<Boolean> val = ClickHouseCast.isTrue(expectedValue);
+        if (!val.isPresent()) {
+            return new ClickHouseUnaryPostfixOperation(randomExpression, ClickHouseUnaryPostfixOperation.ClickHouseUnaryPostfixOperator.IS_NULL, false);
+        }
+        if (val.get()) {
+            return randomExpression;
+        } else {
+            return new ClickHouseUnaryPrefixOperation(randomExpression, ClickHouseUnaryPrefixOperation.ClickHouseUnaryPrefixOperator.NOT);
+        }
+
+    }
+
+    // TODO: refactor others to use this method
+    // https://www.sqlite.org/syntax/ordering-term.html
+    public static String getOrderingTerm(List<ClickHouseSchema.ClickHouseColumn> columns, ClickHouseProvider.ClickHouseGlobalState globalState) {
+        ClickHouseExpression randExpr = new ClickHouseExpressionGenerator(globalState).setColumns(columns)
+                .generateExpression(new ClickHouseSchema.ClickHouseLancerDataType(ClickHouseDataType.Int32));
+        StringBuilder sb = new StringBuilder(ClickHouseVisitor.asString(randExpr));
+        sb.append(" ");
+        if (Randomly.getBoolean()) {
+            sb.append(ClickHouseCommon.getRandomCollate());
+        }
+        if (Randomly.getBoolean()) {
+            if (Randomly.getBoolean()) {
+                sb.append(" ASC");
+            } else {
+                sb.append(" DESC");
+            }
+        }
+        return sb.toString();
+    }
+
+    public static String getIndexedClause(String indexName) {
+        StringBuilder sb = new StringBuilder();
+        if (Randomly.getBoolean()) {
+            sb.append("INDEXED BY ");
+            sb.append(indexName);
+        } else {
+            sb.append("NOT INDEXED");
+        }
+        return sb.toString();
+    }
+
+    public static String getFreeTableName(ClickHouseSchema s) {
```

Comment:
The `getFreeTableName()` and similar methods are now available in https://github.com/sqlancer/sqlancer/blob/master/src/sqlancer/schema/AbstractSchema.java#L63. They should also be available in `ClickhouseSchema`, since it inherits from `AbstractSchema`. Could you please use those instead?

Reply:
Done